src/main/java/com/azure/cosmos/sample/ReadManyItems.java

```diff
@@ -76,178 +79,165 @@ public static void main(String[] args) {
-                ThroughputProperties.createManualThroughput(10000));
-        container = database.getContainer(cosmosContainerResponse.getProperties().getId());
+        System.out.println("Total latency with standard multi-threading: " + totalReadLatency);
+        System.out.println("Total latency using readMany method: " + totalEnhancedReadLatency);
+        System.out.println(
+                "Total request charges with standard multi-threading: " + totalRequestCharges);
+        System.out.println("Total request charges using readMany method: " + totalEnhancedRequestCharges);
     }
 
-    private void createManyItems(final int noOfThreads,
-    final int noOfWritesPerThread) throws Exception {
-        final ExecutorService es = Executors.newCachedThreadPool();
+    private void createManyItems(ArrayList<JsonNode> docs, final int noOfDocs,
+            final int noOfDocsPerThread) throws Exception {
         final long startTime = System.currentTimeMillis();
-        for (int i = 1; i <= noOfThreads; i++) {
-            final Runnable task = () -> {
-                for (int j = 1; j <= noOfWritesPerThread; j++) {
-                    final UUID guid = java.util.UUID.randomUUID();
-                    final String strGuid = guid.toString();
-                    //add the doc id to a global list to be used later for testing read performance.
-                    this.docIDs.add(strGuid);
-                    try {
-                        recordCount.incrementAndGet();                       
-                        Item item = new Item();
-                        item.setId(strGuid);
-                        item.setPk(strGuid);
-                        CosmosItemRequestOptions cosmosItemRequestOptions = new CosmosItemRequestOptions();
-                        container.createItem(item, new PartitionKey(item.getPk()), cosmosItemRequestOptions);
-                        //createItem(item);                                                
-                        insertCount.incrementAndGet();
-                    } catch (final Exception e) {
-                        exceptionCount.incrementAndGet();
-                        System.out.println("Exception: " + e);
-                    }
-                }
-            };
-            es.execute(task);
+        Flux.fromIterable(docs).flatMap(doc -> container.createItem(doc))
+                .flatMap(itemResponse -> {
+                    if (itemResponse.getStatusCode() == 201) {
+                        number_docs_inserted.getAndIncrement();
+                    } else
+                        System.out.println("WARNING insert status code {} != 201" + itemResponse.getStatusCode());
+                    return Mono.empty();
+                }).subscribe(); // ...Subscribing to the publisher triggers stream execution.
+        System.out.println("Doing other things until async doc inserts complete...");
+        while (number_docs_inserted.get() < NUMBER_OF_DOCS) {
```

Comment:
if there is any failure happened for creating item, then this condition will always true.

Reply:
I changed the condition to be based on new value "request_count" which always executes.